### DISCUSSION TITLE
Customize existing User fields?

### DISCUSSION THREAD

**Discussion by Altri** [strapi/strapi#7555] [Legacy Topics]
In the strapi 3.1.4, I'm trying to modify username and email to be not required and Nullable. It has the locked icon and I can't modify it unlike some of the custom fields I added in User. 
I have a custom register mutation that doesn't require the email and username.
I can't seem to find an answer in the existing strapi documentation. Anyone have any idea on how to modify some of the existing fields in user?

![image](https://user-images.githubusercontent.com/9653764/90937641-c786f200-e439-11ea-89d0-5ea978b8de89.png)

**Comment by adrian-ub** [comment]
It may not be recommended but, go to your configuration file in the path
```
extensions/users-permissions/models/User.settings.json
```

And modify the configurable option of the attributes you want to unlock

```
# extensions/users-permissions/models/User.settings.json

# Original
"username": {
       "type": "string",
       "minLength": 3,
       "unique": true,
       "configurable": false,
       "required": true
     },
     "email": {
       "type": "email",
       "minLength": 6,
       "configurable": false,
       "required": true
     },

# Modified
"username": {
       "type": "string",
       "minLength": 3,
       "unique": true,
       "required": true
     },
     "email": {
       "type": "email",
       "minLength": 6,
       "required": true
     },

```

**Reply by derrickmehaffy** [reply]
@Altri the answer provided here is correct and you can change the `required: true` to false, but be aware you may hit issues with register and login if someone tries to login with a username, you will need to apply extensions all over the users-permissions plugin to handle these.
